### PR TITLE
refactor k3s to support swappable components

### PR DESCRIPTION
**Pull request by ShylajaDevadiga**

### CODE REVIEW

File: tests/validation/tests/v3_api/test_import_k3s_cluster.py [hunk 1]
```diff
@@ -16,22 +16,30 @@
...
 RANCHER_INSTANCE_CLASS = os.environ.get("RANCHER_INSTANCE_CLASS",
                                         "db.t2.micro")
 RANCHER_DB_USERNAME = os.environ.get("RANCHER_DB_USERNAME", "adminuser")
 RANCHER_DB_PASSWORD = os.environ.get("RANCHER_DB_PASSWORD", "")
 RANCHER_K3S_KUBECONFIG_PATH = DATA_SUBDIR + "/k3s_kubeconfig.yaml"
+RANCHER_NODE_OS = os.environ.get("RANCHER_NODE_OS", "ubuntu")
+RANCHER_INSTALL_MODE = os.environ.get("RANCHER_INSTALL_MODE", "INSTALL_K3S_VERSION")
+RANCHER_RDS_ENVIRONMENT = os.environ.get("RANCHER_RDS_ENVIRONMENT", "dev")
+RANCHER_RDS_ENGINE_MODE = os.environ.get("RANCHER_RDS_ENGINE_MODE", "")
+RANCHER_CLUSTER_TYPE = os.environ.get("RANCHER_CLUSTER_TYPE", "enternal_db")
```
Comment (bmdepesa) [inline tests/validation/tests/v3_api/test_import_k3s_cluster.py:38]: Typo? `enternal_db`

**Review by bmdepesa — CHANGES_REQUESTED** [review]
Small typo in `RANCHER_CLUSTER_TYPE` default value, then LGTM

File: tests/validation/tests/v3_api/test_import_k3s_cluster.py (hunk 1 above)
Comment (ShylajaDevadiga) [reply tests/validation/tests/v3_api/test_import_k3s_cluster.py]: thanks. fixed

**Review by rancher-max — CHANGES_REQUESTED** [review]
Since we are using `.../terraform/k3s/...` now, can we remove the old files: `.../terraform/master/...` and `.../terraform/worker/...`?

**Review by rancher-max — APPROVED** [review]
LGTM

**Review by bmdepesa — APPROVED** [review]
LGTM

**Pull request merged**